test/unittest/tests/block/test_extent_list_search.cpp

```diff
@@ -0,0 +1,298 @@
+          search_before_after(3 * 4096, &insert_list[1], &insert_list[0]), // At second 12,288.
+          search_before_after(
+            4 * 4096, &insert_list[0], &insert_list[2]), // Between second and third 16,384.
+          search_before_after(5 * 4096, &insert_list[0], &insert_list[2]), // At third 20,480.
+          search_before_after(6 * 4096, &insert_list[2], nullptr),         // After third 24,576.
+        };
+
+        /* Setup. */
+        WT_EXTLIST extlist = {};
+
+        /* Insert extents. */
+        for (const off_size &to_insert : insert_list) {
+            INFO("Insert: {off " << std::showbase << to_insert._off << ", size " << to_insert._size
+                                 << ", end " << to_insert.end() << "}");
+            REQUIRE(__ut_block_off_insert(session, &extlist, to_insert._off, to_insert._size) == 0);
+        }
+
+        extlist_print_off(extlist);
+
+        /* Test. */
+        WT_EXT dummy;
+        uint32_t idx = 0;
+        for (const search_before_after &expected : expected_before_after) {
+            WT_EXT *before = &dummy;
+            WT_EXT *after = &dummy;
+            /* Call. */
+            __ut_block_off_srch_pair(&extlist, expected._off, &before, &after);
+
+            std::ostringstream line_stream;
+            line_stream << "Verify: " << idx << ". off " << expected._off;
+            if (expected._before != nullptr)
+                line_stream << "; Expected: _before: {off " << expected._before->_off << ", size "
+                            << expected._before->_size << ", end " << expected._before->end()
+                            << "}";
+            else
+                line_stream << "; Expected: _before == nullptr";
+
+            if (expected._after != nullptr)
+                line_stream << ", after: {off " << expected._after->_off << ", size "
+                            << expected._after->_size << ", end " << expected._after->end() << "}";
+            else
+                line_stream << ", after == nullptr";
+
+            if (before != nullptr)
+                line_stream << "; Actual: before: {off " << before->off << ", size " << before->size
+                            << ", end " << (before->off + before->size - 1) << "}";
+            else
+                line_stream << "; Actual: before == nullptr";
+
+            if (after != nullptr)
+                line_stream << ", after: {off " << after->off << ", size " << after->size
+                            << ", end " << (after->off + after->size - 1) << "}";
+            else
+                line_stream << ", _after == nullptr";
+
+            std::string line = line_stream.str();
+            INFO(line);
+
+            /* Verify. */
+            if (expected._before != nullptr) {
+                REQUIRE(before != nullptr);
+                REQUIRE(before->off == expected._before->_off);
+                REQUIRE(before->size == expected._before->_size);
+            } else {
```

Comment:
C++ style follows WT style w.r.t. bracketing and therefore single line ifs do not require brackets.

Reply:
Done